unikraft/app/project.go

```diff
@@ -114,6 +115,12 @@ func NewProjectFromOptions(ctx context.Context, opts ...ProjectOption) (Applicat
 		BUILD_DIR: popts.RelativePath(outdir),
 	}
 
+	if _, err := os.Stat(uk.BUILD_DIR); err != nil && os.IsNotExist(err) {
+		if err := os.MkdirAll(uk.BUILD_DIR, 0o755); err != nil {
+			return nil, err
```

Comment:
nit: it's a good practice to wrap some context around the raw error:

```suggestion
			return nil, fmt.Errorf("creating build directory: %w", err)
```

When the user encounters an error, it's easier for them to understand the original intention of the code, as opposed to just `error creating /some/obsure/path`.

It's a very general comment, and I've seen that pattern missing across the code base, but it's never too late to start applying these small qulity of life improvements.